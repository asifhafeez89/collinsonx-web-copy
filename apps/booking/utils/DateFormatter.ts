import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';
import { DATE_TIME_FORMAT } from 'config/Constants';
import 'dayjs/locale/es';
dayjs.extend(utc);
dayjs.extend(timezone);

export function formatDate(
  dateval: Date,
  formattype: string,
  locale?: string
): string {
  dayjs.locale(locale ? locale : 'en');
  return dayjs(dateval).format(formattype);
}

export function formatTimezone(localTimeHour: string, timezone: string) {
  return dayjs.tz(localTimeHour, timezone).format(DATE_TIME_FORMAT);
}
