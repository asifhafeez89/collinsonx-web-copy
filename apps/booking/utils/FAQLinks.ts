export const FAQLink = (accountProvider?: string, locale?: string) => {
  const loungeKey_link_faq = `https://portal.loungekey.com/${locale}/elo/faq/`;
  const priorityPass_link_faq = `https://memberhelp.prioritypass.com/${locale}/support/home`;
  return accountProvider === 'LOUNGE_KEY'
    ? loungeKey_link_faq
    : accountProvider === 'PRIORITY_PASS'
    ? priorityPass_link_faq
    : '';
};
