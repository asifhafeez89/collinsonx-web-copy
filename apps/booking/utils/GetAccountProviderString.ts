export const GetAccountProviderString = (accountProvider?: string) => {
  switch (accountProvider) {
    case 'LOUNGE_KEY':
      return 'Lounge Key';
    case 'PRIORITY_PASS':
      return 'Priority Pass';
    default:
      return 'Priority Pass';
  }
};
