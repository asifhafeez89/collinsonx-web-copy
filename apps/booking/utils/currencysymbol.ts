export const currencyMap: Record<string, string> = {
  GBP: String.fromCharCode(163),
};

export const getCurrencySymbol = (currency: string) =>
  currencyMap[currency] || currency;
