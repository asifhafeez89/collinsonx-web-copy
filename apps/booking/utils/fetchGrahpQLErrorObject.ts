// @ts-ignore
function fetchGrahpQLErrorObject(response, key: string) {
  if (!response && typeof response !== 'object') {
    return null;
  }

  if (!response[key]) {
    return null;
  }

  const error = response[key];

  if (error && 'graphQLErrors' in error) {
    const graphQLErrors = error.graphQLErrors;

    if (graphQLErrors && typeof graphQLErrors === 'object') {
      const data = error.graphQLErrors[0];
      if ('extensions' in data) {
        return data.extensions;
      }
    }
  }

  return null;
}

export default fetchGrahpQLErrorObject;
