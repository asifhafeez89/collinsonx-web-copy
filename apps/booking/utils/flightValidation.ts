const FLIGHT_VALIDATION = /([0-9]{0,1}[a-zA-Z]{1,3})([0-9]*)/gi;

export function validateFlightNumber(
  flight: string
): [boolean, string | undefined, string | undefined] {
  const match = [...flight.matchAll(FLIGHT_VALIDATION)];

  if (!match) {
    return [false, undefined, undefined];
  }

  const carrierLen = match[0]?.[1]?.length ?? 0;
  if (carrierLen < 1 || carrierLen > 3) {
    return [false, undefined, undefined];
  }

  const flightNoLen = match[0]?.[2]?.length ?? 0;
  if (flightNoLen < 1 || flightNoLen > 5) {
    return [false, undefined, undefined];
  }

  return [true, match[0][1], match[0][2]];
}
