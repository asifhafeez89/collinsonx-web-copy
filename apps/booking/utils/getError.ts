import { isMapIterator } from 'util/types';
import { BookingError } from '../constants';

export default function getError(response: any, errorCode: BookingError) {
  if (response && response.errors) {
    return response.errors.find(
      (item: any) => item.extensions.code === errorCode
    );
  }
}
