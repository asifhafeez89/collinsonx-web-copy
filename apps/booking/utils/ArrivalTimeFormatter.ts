import { formatDate } from './DateFormatter';
import { TIME_FORMAT } from 'config/Constants';

export const arrivalTimeFormatter = (
  bookedFrom: string | undefined,
  lastArrival: string | undefined
) =>
  `${formatDate(new Date(`${bookedFrom}`), TIME_FORMAT)}
  -
  ${formatDate(new Date(`${lastArrival}`), TIME_FORMAT)}`;
