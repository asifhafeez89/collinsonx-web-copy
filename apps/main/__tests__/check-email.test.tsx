import * as React from 'react';
import renderer from 'react-test-renderer';
import CheckEmail from '../pages/check-email';
import { MockedProvider } from '@collinsonx/utils/testing';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
    };
  },
}));

describe('<CheckEmail />', () => {
  it('renders', () => {
    const tree = renderer
      .create(
        <MockedProvider mocks={[]} addTypename={false}>
          <CheckEmail />
        </MockedProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
