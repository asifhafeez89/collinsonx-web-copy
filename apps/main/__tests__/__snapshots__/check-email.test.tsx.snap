// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckEmail /> renders 1`] = `
<div
  className="mantine-16jzalw"
/>
`;
